src/libsyntax/parse/parser.rs

```diff
@@ -2663,7 +2665,7 @@ impl<'a> Parser<'a> {
     }
 
     pub fn check_unknown_macro_variable(&mut self) {
-        if self.quote_depth == 0 {
+        if self.quote_depth == 0 && !self.parsing_token_tree {
```

Comment:
It's always `false` because `quote_depth` is never anything other than `0`. IMO, the whole distinction between `$ ident` (two tokens) and `$ident` (one token) is unnecessary and only creates complications.